deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,471 @@
+                pa.array(["foo"] * 10),
+            ],
+            names=["pk_col_1", "sk_col_1", "sk_col_2"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "4-incremental-duplicate-pk": (
+        ["pk_col_1"],
+        [
+            {
+                "key_name": "sk_col_1",
+            },
+            {
+                "key_name": "sk_col_2",
+            },
+        ],
+        [],
+        ["pk_col_1", "sk_col_1", "sk_col_2"],
+        [
+            pa.array([str(i) for i in range(5)] + ["6", "6", "6", "6", "6"]),
+            pa.array([str(i) for i in range(10)]),
+            pa.array(["foo"] * 10),
+        ],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [
+                pa.array([str(i) for i in range(5)] + ["6"]),
+                pa.array([str(i) for i in range(5)] + ["9"]),
+                pa.array(["foo"] * 6),
+            ],
+            names=["pk_col_1", "sk_col_1", "sk_col_2"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+}
+
+
+@pytest.mark.parametrize(
+    [
+        "test_name",
+        "primary_keys",
+        "in_sort_keys",
+        "in_partition_keys",
+        "column_names",
+        "arrow_arrays",
+        "rebase_source_partition_locator",
+        "partition_values",
+        "expected_result",
+        "validation_callback_func",
+        "validation_callback_func_kwargs",
+        "cleanup_prev_table",
+        "use_prev_compacted",
+        "create_placement_group",
+        "records_per_compacted_file",
+        "hash_bucket_count",
+    ],
+    [
+        (
+            test_name,
+            primary_keys,
+            in_sort_keys,
+            in_partition_keys,
+            column_names,
+            arrow_arrays,
+            rebase_source_partition_locator,
+            partition_values,
+            expected_result,
+            validation_callback_func,
+            validation_callback_func_kwargs,
+            cleanup_prev_table,
+            use_prev_compacted,
+            create_placement_group,
+            records_per_compacted_file,
+            hash_bucket_count,
+        )
+        for test_name, (
+            primary_keys,
+            in_sort_keys,
+            in_partition_keys,
+            column_names,
+            arrow_arrays,
+            rebase_source_partition_locator,
+            partition_values,
+            expected_result,
+            validation_callback_func,
+            validation_callback_func_kwargs,
+            cleanup_prev_table,
+            use_prev_compacted,
+            create_placement_group,
+            records_per_compacted_file,
+            hash_bucket_count,
+        ) in INCREMENTAL_TEST_CASES.items()
+    ],
+    ids=[test_name for test_name in INCREMENTAL_TEST_CASES],
+)
+def test_compact_partition_incremental(
```

Comment:
Is testing, rebase and then incremental part of separate PR? I think that is a critical case which must be tested to ensure compactor is logically doing it's job.,

Reply:
Yes - rebase and then incremental will be in a different PR